test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -93,104 +91,520 @@ public void testLogReplicationRoleFlip() throws Exception {
+        log.info("After delta sync, both maps have size {}. Current active corfu[{}] log tail " +
+                        "is {}, standby corfu[{}] log tail is {}", secondBatch, activeClusterCorfuPort,
+                activeRuntime.getAddressSpaceView().getLogTail(), standbyClusterCorfuPort,
+                standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Verify data
+        for (int i = 0; i < secondBatch; i++) {
+            assertThat(mapStandby.containsKey(String.valueOf(i))).isTrue();
+        }
+        log.info("Log replication succeeds without config change!");
+
+        // Perform a role switch
+        corfuStore.tx(DefaultClusterManager.CONFIG_NAMESPACE)
+                .update(DefaultClusterManager.CONFIG_TABLE_NAME, DefaultClusterManager.OP_SWITCH,
+                        DefaultClusterManager.OP_SWITCH, DefaultClusterManager.OP_SWITCH)
+                .commit();
+        assertThat(configTable.count()).isOne();
+
+        // Write 5 more entries to mapStandby
+        for (int i = secondBatch; i < thirdBatch; i++) {
+            standbyRuntime.getObjectsView().TXBegin();
+            mapStandby.put(String.valueOf(i), i);
+            standbyRuntime.getObjectsView().TXEnd();
+        }
+        assertThat(mapStandby.size()).isEqualTo(thirdBatch);
 
-        System.out.println("First batch size is " + firstBatch + ", current standby tail is " +
-                standbyRuntime.getAddressSpaceView().getLogTail() + ", and active tail is " +
-                activeRuntime.getAddressSpaceView().getLogTail());
+        // Wait until data is fully replicated again
+        waitForReplication(size -> size == thirdBatch, mapActive, thirdBatch);
+        log.info("Data is fully replicated again after role switch, both maps have size {}. " +
+                        "Current active corfu[{}] log tail is {}, standby corfu[{}] log tail is {}",
+                thirdBatch, activeClusterCorfuPort, activeRuntime.getAddressSpaceView().getLogTail(),
+                standbyClusterCorfuPort, standbyRuntime.getAddressSpaceView().getLogTail());
+
+        // Double check after 10 seconds
+        TimeUnit.SECONDS.sleep(mediumInterval);
+        //assertThat(mapActive.size()).isEqualTo(thirdBatch);
+        waitForReplication(size -> size == thirdBatch, mapActive, thirdBatch);
+        assertThat(mapStandby.size()).isEqualTo(thirdBatch);
+    }
 
-        ExecutorService executorService = Executors.newFixedThreadPool(2);
+    /**
+     * This test verify config change with a role switch during a snapshot sync transfer phase.
+     * <p>
+     * 1. Init with corfu 9000 active and 9001 standby
+     * 2. Write 50 entries to active map
+     * 3. Start log replication: Node 9010 - active, Node 9020 - standby
+     * 4. Perform a role switch with corfu store
+     * 5. Standby will drop messages and keep size 0
+     * 6. Verify active map becomes size 0, since source size is 0
+     */
+    //@Test
```

Comment:
Since there is no way to make sure apply completes before the writer reset, we do not support DuringTransferPhase & DuringApplyPhase now.